### PR TITLE
AAE-16959 Extend IdentityLink Api with user identity details attribute

### PR DESCRIPTION
**Pull request by igdianov**
This PR extends IdentityLink api to be able to store and query implementation specific user identity details in the process and task entity contexts. 

Part of: https://alfresco.atlassian.net/browse/AAE-16959

### CODE REVIEW

**Comment by balsarori** [comment]
I'm not sure it's a good idea to save such info in the engine level. The engine's concern should concentrate on process execution. Can't this logic and data live somewhere outside of the engine?

**Comment by igdianov** [comment]
> I'm not sure it's a good idea to save such info in the engine level. The engine's concern should concentrate on process execution. Can't this logic and data live somewhere outside of the engine?

The info is needed to be attached to the process execution and task entities via identity links, so that we can generate audit events with actor details from identity service. The identity link is the best fit for it by extending the details attribute, so we don't have to introduce any dependency on Hibernate/JPA/Liquibase in the Activiti Cloud Process Runtime.

File: activiti-core/activiti-engine/pom.xml [hunk 1]
```diff
@@ -201,6 +216,33 @@
...
+            <phase>compile</phase>
+            <goals>
+              <goal>unpack</goal>
+            </goals>
+            <configuration>
+              <artifactItems>
+                <artifactItem>
+                  <groupId>org.activiti</groupId>
+                  <artifactId>activiti-engine</artifactId>
+                  <version>7.2.0</version>
```
Comment (graric) [inline activiti-core/activiti-engine/pom.xml]: What do you think about using a variable with a meaningful name(e.g. activiti.previous.version)?

**Review by graric — APPROVED** [review]

File: activiti-core/activiti-engine/src/main/resources/org/activiti/db/upgrade/activiti.mysql55.upgradestep.800.to.810.history.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+alter table ACT_HI_IDENTITYLINK add column DETAILS_ LONGBLOB;
```
Comment (erdemedeiros) [inline activiti-core/activiti-engine/src/main/resources/org/activiti/db/upgrade/activiti.mysql55.upgradestep.800.to.810.history.sql:1]: Missing migration from `mysql`? I only see `mysql55`

File: activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/db/DbSchemaUpdateTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        CommandExecutor commandExecutor = processEngine.getProcessEngineConfiguration().getCommandExecutor();
+        CommandConfig config = new CommandConfig().transactionNotSupported();
+
+        // and when
+        PropertyEntity schemaVersion = commandExecutor.execute(config,
+            commandContext -> commandContext.getDbSqlSession()
+                .selectById(PropertyEntity.class,"schema.version"));
+
+        // then
+        assertEquals(ProcessEngine.VERSION, schemaVersion.getValue());
```
Comment (erdemedeiros) [inline activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/db/DbSchemaUpdateTest.java]: Please, use `AssertJ` assertions

File: activiti-core/activiti-engine/src/test/java/org/activiti/engine/test/history/HistoricTaskInstanceTest.java [hunk 4]
```diff
@@ -605,6 +604,62 @@ public void testHistoricIdentityLinksOnTask() throws Exception {
...
+                    assertThat(link.getDetails()).isEqualTo("details".getBytes());
+                    foundCustom = true;
+                }
+            }
+        }
+
+        assertThat(foundAssignee).isTrue();
+        assertThat(foundCandidateGroup).isTrue();
+        assertThat(foundCandidateUser).isTrue();
+        assertThat(foundCustom).isTrue();
```
Comment (erdemedeiros) [inline activiti-core/activiti-engine/src/test/java/org/activiti/engine/test/history/HistoricTaskInstanceTest.java]: All this block can be simplified to
```suggestion
        assertThat(historicIdentityLinks)
            .extracting(
                HistoricIdentityLink::getType,
                HistoricIdentityLink::getGroupId,
                HistoricIdentityLink::getUserId,
                HistoricIdentityLink::getDetails)
            .containsExactlyInAnyOrder(
                tuple("candidate", "sales", null, null),
                tuple("candidate", null, "fozzie", null),
                tuple("assignee", null, "kermit", null),
                tuple("actor", null, "gonzo", "details".getBytes())
            );
```

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Apart from some small details, LGTM

File: activiti-core/activiti-engine/src/main/resources/org/activiti/db/upgrade/activiti.mysql55.upgradestep.800.to.810.history.sql (hunk 2 above)
Comment (igdianov) [reply activiti-core/activiti-engine/src/main/resources/org/activiti/db/upgrade/activiti.mysql55.upgradestep.800.to.810.history.sql:1]: Mysql updgrade exists here: https://github.com/Activiti/Activiti/pull/4444/files#diff-00b1b0e8c4ec776b678c0eea023a4e9faad83407da3057d95e46344d37323df3R1

Comment (erdemedeiros) [reply activiti-core/activiti-engine/src/main/resources/org/activiti/db/upgrade/activiti.mysql55.upgradestep.800.to.810.history.sql:1]: The one you mentioned is `mssql`, not `mysql` :D

**Review by erdemedeiros — APPROVED** [review]

**Pull request merged**